sources/PTYSession.m

```diff
@@ -3061,29 +3066,26 @@ - (void)setBlend:(float)blendVal {
     [_textview setBlend:blendVal];
 }
 
-- (BOOL)antiIdle
-{
+- (BOOL)antiIdle {
     return _antiIdleTimer ? YES : NO;
 }
 
-- (void)setAntiIdle:(BOOL)set
-{
-    if (set == [self antiIdle]) {
-        return;
+- (void)setAntiIdle:(BOOL)set {
+    [_antiIdleTimer invalidate];
+    [_antiIdleTimer release];
+    _antiIdleTimer = nil;
+    
+    // FIXME: corner case: part 1of2 of a kludgey way to prevent a flood when period is set to 1
+    if (_antiIdlePeriod > 0 && _antiIdlePeriod < 1.1) {
+        _antiIdlePeriod = 1.1;
     }
-
-    if (set) {
-        NSTimeInterval period = MIN(60, [iTermAdvancedSettingsModel antiIdleTimerPeriod]);
-
-        _antiIdleTimer = [[NSTimer scheduledTimerWithTimeInterval:period
+    
+    if (set && _antiIdlePeriod > 0) {
+        _antiIdleTimer = [[NSTimer scheduledTimerWithTimeInterval:_antiIdlePeriod
```

Comment:
I'll do it of course, but for the record my reasoning was:

I am trying to avoid any arbitrary limits like that. It's not that I can think of a valid use-case for sending a continuous stream of nulls every 0.1 second, it's that it's none of my business to say 2 seconds is the shortest anyone else needs any more than 60 seconds was.

And the min would have to be 2 seconds to avoid that flood/race condition, which as a user just seems like an inexplicably arbitrary number. Taking 1 and internally nudging it up to 1.1 hides the problem and still works the same as 1.

Really I wanted to allow less than 1 second but I just couldn't figure out how to get it to accept floating point values, and whole seconds and a minimum of 1 second are both actually perfectly reasonable so I didn't pursue it.

I can see dispensing with the 0 option, because it makes more than one way to disable the feature. You have clicked the checkbox to enable sending, and yet nothing is sending. ok I get that.

2 seconds is actually well below my own lowest real value which is 10 seconds. Some connections are as aggressive as less than 1 minute, and you have to account for the fact that on some connections, 1 or 2 or even more packets might be lost and not make it to the aggressive component, and so in the worst cases you might need to send maybe every 10 seconds to ensure that you get 3 chances to get at least 1 packet all the way through within 30 seconds. So this still works for me, it's just slightly hokier to me from the users perspective. Why does it bottom out at 2?